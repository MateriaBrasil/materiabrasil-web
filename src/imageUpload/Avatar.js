import React from 'react'
import Link from 'react-router-dom/Link'

import Image from './Image'

export default props =>
  props.editable ? (
    <Link to={props.editable}>
      <Image {...props} />
    </Link>
  ) : (
    <Image {...props} />
  )
