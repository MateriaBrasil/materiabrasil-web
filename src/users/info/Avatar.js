import React from 'react'
import Link from 'react-router-dom/Link'

import Image from './Image'

export default props => {
  const { editable } = props

  if (!editable) {
    return <Image {...props} />
  }

  return (
    <Link to="/profile/avatar">
      <Image {...props} />
    </Link>
  )
}
