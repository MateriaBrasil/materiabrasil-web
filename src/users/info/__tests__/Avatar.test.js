import React from 'react'
import renderer from 'react-test-renderer'

import Avatar from '../Avatar'

jest.mock('react-router-dom/Link', () => props => (
  <div {...props}>{props.children}</div>
))

jest.mock('../Image', () => props => <div {...props}>Image</div>)

describe('when editable', () => {
  it('renders correctly', () => {
    const props = { editable: true, bar: 'foo' }
    const tree = renderer.create(<Avatar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('when not editable', () => {
  it('renders correctly', () => {
    const props = { editable: false, bar: 'foo' }
    const tree = renderer.create(<Avatar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
