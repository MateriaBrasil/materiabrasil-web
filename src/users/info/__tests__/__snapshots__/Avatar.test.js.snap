// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when editable renders correctly 1`] = `
<div
  to="/profile/avatar"
>
  <div
    bar="foo"
    editable={true}
  >
    Image
  </div>
</div>
`;

exports[`when not editable renders correctly 1`] = `
<div
  bar="foo"
  editable={false}
>
  Image
</div>
`;
